syft/rekor/queryRekor/utils.go

```diff
@@ -0,0 +1,274 @@
+	"github.com/sigstore/rekor/pkg/generated/models"
+	"github.com/spdx/tools-golang/spdx"
+	"github.com/spdx/tools-golang/tvloader"
+)
+
+var trustedCertIdentities = []cosign.Identity{
+	{Subject: "mdeicas@google.com", Issuer: "sigstore.dev"},
+	{Subject: "mdeicas@google.com", Issuer: "sigstore"},
+}
+
+// TODO: identities list is not working. Needs to work to verify github action runs because they don't have an email field
+var certCheckOpts = &cosign.CheckOpts{
+	//Identities: trustedCertIdentities,
+	//CertEmail: "mdeicas@google.com",
+
+	// Why is there an option to prove sigVerifier?
+	// addresses after SAN. what is this and should we check it
+
+}
+
+type spdxSbomWithMetadata struct {
+	sbom        spdx.Document2_2
+	packageName string
+	namespace   string // i.e.  uri
+}
+
+type digest struct {
+	Sha256 string
+}
+
+type subject struct {
+	Name   string
+	Digest digest
+}
+
+type lumjjbSbomType struct {
+	Format string
+	Digest digest
+	Uri    string
+}
+
+type buildMetadata struct {
+	ArtifactSourceRepo             string `json:"artifact-source-repo"`
+	ArtifactSourceRepoCommit       string `json:"artifact-source-repo-commit"`
+	AttestationGeneratorRepo       string `json:"attestation-generator-repo"`
+	AttestationGeneratorRepoCommit string `json:"attestation-generator-repo-commit"`
+}
+
+type lumjjbPredType struct {
+	Sboms         []lumjjbSbomType
+	BuildMetadata buildMetadata `json:"build-metadata"`
+}
+
+type inTotoAttestation struct {
+	AttType       string `json:"_type"`
+	PredicateType string
+	Subject       []subject
+	Predicate     lumjjbPredType
+}
+
+func pprintCert(cert *x509.Certificate) {
+
+	s := fmt.Sprint(
+		"\n*** certificate *** \n",
+		"subject:", cert.Subject.CommonName,
+		"\nsubject alternative name: email: ", cert.EmailAddresses,
+		"\nissuer: ", cert.Issuer.CommonName,
+		"\nnotBefore: ", cert.NotBefore.String(),
+		"\nnotAfter: ", cert.NotAfter.String(),
+		"\n*** certificate *** \n\n",
+	)
+	fmt.Println(s)
+}
+
+func pprintStruct(v any) {
+	json, err := json.MarshalIndent(v, "", "\t")
+	if err != nil {
+		fmt.Println(err)
+	}
+	fmt.Println(string(json))
+}
+
+// decode and parse certificate
+func parseCert(decodedCert string) (*x509.Certificate, error) {
+
+	pem, _ := pem.Decode([]byte(decodedCert))
+	if pem == nil {
+		return nil, errors.New("certificate could not be found")
+	}
+
+	cert, err := x509.ParseCertificate(pem.Bytes)
+	if err != nil {
+		log.Debug("Error parsing x509 certificate")
+		return nil, err
+	}
+	return cert, err
+}
+
+// decode and marshall log entry body
+func parseEntry(entry *models.LogEntryAnon) (*models.IntotoV001Schema, error) {
+
+	bodyEncoded, ok := entry.Body.(string)
+	if !ok {
+		return nil, errors.New("attempted to parse entry body as string, but failed")
+	}
+
+	bodyDecoded, err := base64.StdEncoding.DecodeString(bodyEncoded)
+	if err != nil {
+		log.Debug("Error base64 decoding body")
+		return nil, err
+	}
+
+	var intoto *models.Intoto = new(models.Intoto)
```

Comment:
nit: usually - for 1 reason or another, people have strayed away from new and instead used this instead.

```
intoto := &models.Intoto{}
```